com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java

```diff
@@ -549,34 +568,34 @@ private static Options getCommandLineOptions() {
         return options;
     }
 
-    private static CommandLine parse(String[] args) {
+    private static CommandLine parse(String[] args) throws OptionValidationException {
         Options options = getCommandLineOptions();
 
         CommandLineParser parser = new PosixParser();
-        CommandLine cmd = null;
+        CommandLine cmd;
         try {
             cmd = parser.parse(options, args);
         } catch (ParseException e) {
             System.err.println(e.getMessage());
-            System.exit(5);
+            throw new OptionValidationException(5);
         }
         if (cmd.hasOption("h")) {
-            HelpFormatter helpFormatter = new HelpFormatter( );
+            HelpFormatter helpFormatter = new HelpFormatter();
             helpFormatter.printHelp("bob [options] [commands]", options);
-            System.exit(0);
+            return null;
         }
         if (cmd.hasOption("ce") || cmd.hasOption("pk")) {
             System.out.println("Android signing using certificate and private key is no longer supported. You must use keystore signing.");
             HelpFormatter helpFormatter = new HelpFormatter( );
             helpFormatter.printHelp("bob [options] [commands]", options);
-            System.exit(1);
+            throw new OptionValidationException(1);
         }
 
         return cmd;
     }
 
-    private static Project createProject(String rootDirectory, String buildDirectory, String email, String auth) {
-        Project project = new Project(new DefaultFileSystem(), rootDirectory, buildDirectory);
+    private static Project createProject(ClassLoader classLoader, String rootDirectory, String buildDirectory, String email, String auth) {
+        Project project = new Project(classLoader, new DefaultFileSystem(), rootDirectory, buildDirectory);
```

Comment:
The editor sets its own classloader, while the bob doesn't specify any. Made it an arg.